src/main/java/com/joss/conductor/mobile/annotation/PlatformIdHelper.java

```diff
@@ -0,0 +1,53 @@
+package com.joss.conductor.mobile.annotation;
+
+import com.joss.conductor.mobile.Platform;
+import org.openqa.selenium.By;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Field;
+
+/**
+ * Created on 3/14/18.
+ */
+public final class PlatformIdHelper {
+
+    public static void initIds(Object page, Platform platform) throws Exception {
+
+        Class current = page.getClass();
+        do {
+            for (Field field : current.getDeclaredFields()) {
+                for (Annotation annotation : field.getDeclaredAnnotations()) {
+                    if (!field.getType().equals(By.class)) {
+                        throw new ClassCastException("Must be of type By! Field name: " + field.getName());
+                    }
+
+                    if (annotation instanceof AndroidId) {
+                        AndroidId androidIdAnnotation = (AndroidId) annotation;
+                        if (platform.equals(Platform.ANDROID)) {
```

Comment:
I'd just move this `if` statement into the above `if` statement and just add an `&&` because you essentially _only_ want to do this if both of these things are true.